src/TestDox/TestDoxResultsMerger.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ParaTest\TestDox;
+
+use PHPUnit\Logging\TestDox\TestResult as TestDoxTestMethod;
+use PHPUnit\Logging\TestDox\TestResultCollection;
+use PHPUnit\Logging\TestDox\TestResultCollection as TestdoxTestResultCollection;
+use ReflectionException;
+use ReflectionMethod;
+use SplFileInfo;
+
+use function array_merge;
+use function assert;
+use function file_get_contents;
+use function is_subclass_of;
+use function ksort;
+use function uksort;
+use function unserialize;
+use function usort;
+
+/** @internal */
+final readonly class TestDoxResultsMerger
+{
+    /**
+     * @param list<SplFileInfo> $testdoxFiles
+     *
+     * @return array<string,TestdoxTestResultCollection>
+     */
+    public function getResultsFromTestdoxFiles(array $testdoxFiles): array
+    {
+        /** @var array<string,TestdoxTestResultCollection> $testMethodsGroupedByClass */
+        $testMethodsGroupedByClass = [];
+        foreach ($testdoxFiles as $testdoxFile) {
+            $testdoxFileContents = file_get_contents($testdoxFile->getPathname());
```

Comment:
```suggestion
            if (! $testdoxFile->isFile()) {
                continue;
            }
            $testdoxFileContents = file_get_contents($testdoxFile->getPathname());
```
This assumption is wrong: if the number of tests is lower than the available processes, the Nth file is never written so it doesn't exists.